### ISSUE TITLE
[BUG] Unsupported reference object in parameters schema

### ISSUE THREAD

**Issue by sergiopatricio** [rswag/rswag#563]
## Describe the bug
With the update to `2.6.0`, I have some tests failing. The reason for the failure is the way schemas are resolved in parameters.

Reading https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#parameterObject it says that `schema` can be a "Schema Object" or a "Reference Object".
I think the code on `rswag/specs/request_factory.rb` is not prepared for the "Reference Object" option.

## Steps to Test or Reproduce
With the following defined:
```
components:
  schemas:
    StateEnum:
      type: string
      enum:
      - state1
      - state2
```

Add a parameter with:
```
parameter name: :state,
          in: :query,
          required: false,
          schema: { '$ref' => '#/components/schemas/StateEnum' }
```

When using the parameter the following error is raised:
```
NoMethodError:
   undefined method `to_sym' for nil:NilClass

             case param[:schema][:type].to_sym
                                       ^^^^^^^
# rswag-specs-2.6.0/lib/rswag/specs/request_factory.rb:131:in `build_query_string_part'
# rswag-specs-2.6.0/lib/rswag/specs/request_factory.rb:118:in `block (2 levels) in add_path'
# rswag-specs-2.6.0/lib/rswag/specs/request_factory.rb:116:in `each'
# rswag-specs-2.6.0/lib/rswag/specs/request_factory.rb:116:in `each_with_index'
# rswag-specs-2.6.0/lib/rswag/specs/request_factory.rb:116:in `block in add_path'
```

Looking through `rswag/specs/request_factory.rb`, when expanding parameters, it is prepared to reference a parameter with `$ref`, but is not prepared to expand a referenced schema inside a parameter. So `type` (and other keys) were not expanded in the schema containing the `$ref`.

## Expected behavior
No error should be raised.

## Rswag Version
2.6.0

## Relates to which version of OAS (Open API Specification)
- [ ] OAS2
- [x] OAS3
- [ ] OAS3.1

**Comment by sergiopatricio** [comment]
The following may also raise the same error:
```
parameter name: :something,
          in: :query,
          schema: {
            oneOf: [
              { type: :string },
              { type: :integer },
            ],
          }
```

**Comment by sergiopatricio** [comment]
A quick fix can be something like
```diff
--- a/rswag-specs/lib/rswag/specs/request_factory.rb
+++ b/rswag-specs/lib/rswag/specs/request_factory.rb
@@ -145,6 +145,8 @@ module Rswag

         # OAS 3: https://swagger.io/docs/specification/serialization/
         if swagger_doc && doc_version(swagger_doc).start_with?('3') && param[:schema]
+          return "#{name}=#{value}" unless param[:schema].key?(:type)
+
           style = param[:style]&.to_sym || :form
           explode = param[:explode].nil? ? true : param[:explode
```

or even just
```diff
--- a/rswag-specs/lib/rswag/specs/request_factory.rb
+++ b/rswag-specs/lib/rswag/specs/request_factory.rb
@@ -148,7 +148,7 @@ module Rswag
           style = param[:style]&.to_sym || :form
           explode = param[:explode].nil? ? true : param[:explode]

-          case param[:schema][:type].to_sym
+          case param[:schema][:type]&.to_sym
           when :object
             case style
             when :deepObject
```

**Comment by sergiopatricio** [comment]
Fixed in #564.
